### PR TITLE
Improve consensus unit tests

### PR DESCRIPTION
**Pull request by metajack**
Consensus is built on the Tokio executor for futures, but currently uses `futures::executor` all over the place. This was discovered by trying to improve the test timeout behavior and realizing that the futures were not executing in a Tokio runtime.

This PR does the following:
- Changes tests using `futures::executor::block_on` to use `tokio::runtime::Runtime::block_on` instead.
- Uses `async` blocks instead of `async move` blocks, as the `Runtime` must be dropped from outside a future as it calls block_on in the drop logic. The `async move` was moving the `Runtime` inside the block, which was then dropped inside the block, causing a panic for invoking runtime functions inside a runtime (similar to a double borrow).
- Removes the crash handler, which force exited the process on panic, which is not the behavior the test harness is designed for.
- Replaces the timeout panic with `tokio::time::timeout` for more graceful timeout handling.

This PR does not remove all instances of `futures::executor`, but that should be done eventually.

### CODE REVIEW

**Comment by metajack** [comment]
cc @davidiw @rexhoffman @bmwill @aching

File: consensus/src/chained_bft/liveness/pacemaker_test.rs [hunk 1]
```diff
@@ -28,12 +32,13 @@ fn test_pacemaker_time_interval() {
 #[test]
 /// Verify that Pacemaker properly outputs local timeout events upon timeout
 fn test_basic_timeout() {
+    let mut runtime = consensus_runtime();
```
Comment (zekun000) [inline consensus/src/chained_bft/liveness/pacemaker_test.rs]: this probably can be simple tokio::test

File: consensus/src/chained_bft/liveness/proposal_generator_test.rs [hunk 2]
```diff
@@ -38,20 +38,24 @@ fn test_proposal_generation_empty_tree() {
     let genesis = block_store.root();
 
     // Generate proposals for an empty tree.
-    let proposal_data =
-        block_on(proposal_generator.generate_proposal(1, minute_from_now())).unwrap();
+    let proposal_data = runtime
+        .block_on(proposal_generator.generate_proposal(1, minute_from_now()))
```
Comment (zekun000) [inline consensus/src/chained_bft/liveness/proposal_generator_test.rs]: ditto

File: consensus/src/chained_bft/liveness/pacemaker_test.rs (hunk 1 above)
Comment (bmwill) [reply consensus/src/chained_bft/liveness/pacemaker_test.rs]: `consensus_runtime` does a few other things like setting up a logger if --nocapture is passed so while we probably could move to using `tokio::test` we'd loose that behavior. Another issue would be if any code expects to have a handle directly onto a runtime (which you can't get with the test macro) Course there's nothing stopping us from building our own macro to just do that stuff too.

Comment (metajack) [reply consensus/src/chained_bft/liveness/pacemaker_test.rs]: These tests didn't have a runtime before my change, so in the end the behavior is unchanged. I agree though, and as soon as we can get consensus more cleaned up, we can put a #[consensus::test] macro to use that does all these nice things.

**Pull request closed without merging**